weaver/processes/ogc_api_process.py

```diff
@@ -1,24 +1,65 @@
 from typing import TYPE_CHECKING
 
+from weaver.processes.constants import PACKAGE_FILE_TYPE, JobInputsOutputsSchema
+from weaver.processes.convert import convert_input_values_schema, convert_output_params_schema
 from weaver.processes.wps_process_base import OGCAPIRemoteProcessBase, RemoteJobProgress
 from weaver.status import Status
 
 if TYPE_CHECKING:
-    from weaver.typedefs import JSON, UpdateStatusPartialFunction
+    from typing import Optional
+
+    from weaver.typedefs import (
+        CWL_ExpectedOutputs,
+        ExecutionInputsMap,
+        ExecutionOutputsMap,
+        JobInputs,
+        JobOutputs,
+        JSON,
+        UpdateStatusPartialFunction
+    )
     from weaver.wps.service import WorkerRequest
 
 
 class OGCAPIRemoteProcess(OGCAPIRemoteProcessBase):
     process_type = "OGC API"
 
-    def __init__(self,
-                 step_payload,      # type: JSON
-                 process,           # type: str
-                 request,           # type: WorkerRequest
-                 update_status,     # type: UpdateStatusPartialFunction
-                 ):                 # type: (...) -> None
+    def __init__(
+        self,
+        step_payload,   # type: JSON
+        process,        # type: str
+        request,        # type: Optional[WorkerRequest]
+        update_status,  # type: UpdateStatusPartialFunction
+    ):                  # type: (...) -> None
         super(OGCAPIRemoteProcess, self).__init__(step_payload, process, request, update_status)
-        self.url = process
         self.provider, self.process = process.rsplit("/processes/", 1)
-        self.update_status(f"Provider {self.provider} is selected for process [{self.process}].",
+        self.url = self.provider  # dispatch operation re-aggregates the provider with the necessary endpoints
+        self.update_status(f"Provider [{self.provider}] is selected for process [{self.process}].",
                            RemoteJobProgress.SETUP, Status.RUNNING)
+
+    def prepare(self, workflow_inputs, expected_outputs):
+        # type: (JobInputs, CWL_ExpectedOutputs) -> None
+
+        # by-ref update of "CWL expected outputs" since they are not returned
+        # the 'type' is expected by the output staging operation to retrieve a file as remote URI
+        # any other 'type' does nothing, but a valid CWL type is needed to avoid raising the validation
+        for output in expected_outputs.values():
+            cwl_type = PACKAGE_FILE_TYPE if "format" in output else "string"
+            output.setdefault("type", cwl_type)
+
+    def format_inputs(self, job_inputs):
+        # type: (JobInputs) -> ExecutionInputsMap
+        inputs = convert_input_values_schema(job_inputs, JobInputsOutputsSchema.OGC)
+        return inputs
```

Comment:
Why use the same `JobInputsOutputsSchema ` for input and output while each have different converting function `convert_input_values_schema` and `convert_output_params_schema`

Reply:
The schema represents the use of list vs mapping representation. This is identical on both side. However, the "inputs" in this case are the submitted data, while outputs are the desired value/ref/format definitions. The 2 functions handle this content differently and separately.